### PR TITLE
DX - 77: Added the missing modules handle function, upgraded uuid and version bump

### PR DESCRIPTION
**Pull request by cs-raj**
version bump:
@contentstack/cli: 1.20.0 -> 1.21.0
@contentstack/cli-migration: 1.5.6 -> 1.6.0

### CODE REVIEW

File: packages/contentstack-migration/src/utils/modules.js [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+  });
+
+  return [...dependencies];
+}
+
+function createPackageJson(directory) {
+  const templateString = `{
+    "name": "MigrationPackage",
+    "version": "1.0.0",
+    "main": "1.js",
```
Comment (aman19K) [inline packages/contentstack-migration/src/utils/modules.js:84]: @cs-raj , any specific reason for mentioning 1.js in main.

**Review by aman19K — CHANGES_REQUESTED** [review]

File: packages/contentstack-migration/src/utils/modules.js (hunk 1 above)
Comment (cs-raj) [reply packages/contentstack-migration/src/utils/modules.js:84]: No, It was just the file with which it was added, shall I remove it?

Comment (cs-raj) [reply packages/contentstack-migration/src/utils/modules.js]: Removed

File: packages/contentstack-migration/src/utils/modules.js [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+  'trace_events',
+  'tty',
+  'url',
+  'util',
+  'v8',
+  'vm',
+  'wasi',
+  'worker_threads',
+  'zlib',
+]);
```
[the remark this answers is not in the document]
Comment (cs-raj) [reply packages/contentstack-migration/src/utils/modules.js]: Thanks, Used the second one

**Review by aman19K — DISMISSED** [review]

File: packages/contentstack-migration/src/utils/modules.js [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+    "keywords": [],
+    "author": "",
+    "license": "ISC",
+    "description": ""
+  }`;
+
+  fs.writeFileSync(path.join(sanitizePath(directory), 'package.json'), templateString);
+}
+
+function installDependencies(dependencies, directory) {
```
Comment (shafeeqd959) [inline packages/contentstack-migration/src/utils/modules.js:57]: How do we get the version of a dependency ? Installing the latest version of a dependency would be an issue here ??

File: packages/contentstack-migration/src/utils/modules.js [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+      createPackageJson(dirPath);
+    }
+
+    installDependencies(dependencies, dirPath);
+  } else {
+    console.log(`You don't have write permission to the directory`);
+    return false;
+  }
+
+  console.log(`All modules installed successfully`);
```
Comment (shafeeqd959) [inline packages/contentstack-migration/src/utils/modules.js:107]: "All dependencies installed successfully"

File: packages/contentstack-migration/src/commands/cm/stacks/migration.js [hunk 5]
```diff
@@ -129,6 +129,10 @@ class MigrationCommand extends Command {
     set(MANAGEMENT_SDK, mapInstance, stackSDKInstance);
     set(MANAGEMENT_CLIENT, mapInstance, APIClient);
 
+    if (!(await installModules(filePath, multi))) {
+      this.log(`Cannot proceed with migration as there will be module not found issue as package.json is not created`);
```
Comment (shafeeqd959) [inline packages/contentstack-migration/src/commands/cm/stacks/migration.js:133]: "Failed to install the dependencies of the given scripts."  please change the text , then we can exit

**Review by shafeeqd959 — DISMISSED** [review]

File: packages/contentstack-migration/src/utils/modules.js (hunk 3 above)
Comment (cs-raj) [reply packages/contentstack-migration/src/utils/modules.js:57]: We don't get the versions of the dependency we try to install the latest version for our use case of entry migration that won't be a problem untill we release major version. Asking or letting the user specify specific version will be as good as creating package.json manually or running npm i with specific version. The issue might arise with user running their own script with specific version

File: packages/contentstack-migration/src/utils/modules.js (hunk 4 above)
Comment (cs-raj) [reply packages/contentstack-migration/src/utils/modules.js:107]: Got it I will add it

File: packages/contentstack-migration/src/commands/cm/stacks/migration.js (hunk 5 above)
Comment (cs-raj) [reply packages/contentstack-migration/src/commands/cm/stacks/migration.js]: added

**Review by aman19K — APPROVED** [review]

**Pull request merged**